portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,72 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  private static final String COMMENT_INPUT = "comment-input";
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Arjavi!</h1>");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
```

Comment:
(optional) just for discussion
Since you are declaring Comment class, it is actually possible to put this into that class as a static function getLatestComments.
and another static function to insertComment.
if you want, these query functions can also be put into a nested class and finally the datastore can be moved there as well

Reply:
Thanks for the suggestion! I have created a separate CommentsManager class to do this